README.md

```diff
@@ -158,6 +176,10 @@ class MyComponent extends \Keboola\Component\BaseComponent
 
 If any constraint of config definition is not met a `UserException` is thrown. That means you don't need to handle the messages yourself. 
 
+## Migration from version 6 to version 7
+
+The default entrypoint of component (in `index.php`) changed from `BaseComponent::run()` to `BaseComponent::execute()`. While running the component via `run` method is still supported, you need to use `execute()` if you want to take advantage of sync action support. 
+
```

Comment:
to je zoufale nahovno bcbreak. run.php se testuje zpravidla jen integracnima testama a ty zase zpravidla netestujou sync akce. to znamena, ze kdyz updatnes komponentu a zapomenes ji zmenit run.php tak testy projdou, ale sync akce nebudou fungovat. 

premyslim jak z toho vybruslit a nejjednodussi se mi zda, prejmenovat `run` na `runAction`, coz se me sice nelibi, protoze se pak trosku zbytecne bude plest se sync akcema, ale udela to poradnej bc break, kterej zajisti, ze se pokazenej `run.php` nedostane do produkce.

Reply:
taky bych do bc breaku pridal, ze testovaci konfigurace musi mit run, v testech to casto chybi, protoze to melo default hodnotu

Reply:
Myšlenka byla, že když na tom teď závisíme (předtím jsme nikde v kódu run metodu nevyžadovali) tak bysme měli run vyžadovat. 

Ale na druhou stranu si říkám, že by asi bylo lepší ten BC break udělat co nejmenší - aby to byl co nejmenší pain upgradovat.

Reply:
> prejmenovat run na runAction

Což ale neudělá nic. `MyComponent::run()` furt existuje a furt se volá z indexu. A nemusí volat parent::run(), takže ani kdybych tam dal exceptionu, tak to nepomůže. 

Napadá mě v konstruktoru přes reflexi zkontrolovat, že run není public ale protected. Když bude protected, tak nejde zavolat z indexu a problem solved. A komponenty určitě parent constructor volají. Jinak by přišli o většinu cool věcí, co component umí :)

Reply:
Zkusím to vykopnout a pushnout.

Reply:
36bfecd

Reply:
no tak z teho komentara su jelen :)

> A nemusí volat parent::run(), 
> A komponenty určitě parent constructor volají.

Reply:
z komentare jsem jelen, ale myslim, ze by to tak mohlo byt

Reply:
Ve zkratce:

1. v `index.php` mám `$myComponent->run()`
2. přejmenuju `Component::run()` na `Component::runAction()`
3. `MyComponent::run` se nepřejmenuje, takže volání v 1. stále proběhne jako by nic

Oproti tomu, když něco přidám do konstruktoru, tak tam je slušná jistota, že se to zavolá i v poděděné komponentě. 

To jsem tím chtěl říct.

Reply:
> MyComponent::run se nepřejmenuje, takže volání v 1. stále proběhne jako by nic 

neprobehene, protoze neexistuje parent::run()

Reply:
Což předpokládá, že v metodě mycomponent::run se zavolá parent::run. Což myslím, že [se moc neděje](https://github.com/search?p=2&q=org%3Akeboola+%22parent%3A%3Arun%22&type=Code).